### ISSUE TITLE
Subtask via VTODO RELATED-TO (as Nextcloud / ownCloud / ...)

### ISSUE THREAD

**Issue by Matthias84** [dmfs/opentasks#341]
Currently subtasks are implemented as list (see #37) which is pretty nice. But as companion client which interacts with a Nextcloud server and Task app, this causes problems:
* Subtasks created via the webfrontend will become an main task at opentasks.
* Viceversa subtasks created with OT will just show up at the description field at the webfrontend.

I can deal with it, but a perfect workflow would present all infos in the same way :smile: 

Technically the iCalendar / calDAV elements are referenced [via RELATED-TO attributes](https://tools.ietf.org/html/rfc5545#section-3.8.4.5) to created a hierachy. OpenTask would need to understand this model and (maybe later) present also subtasks at the widget etc...

**Comment by jkufner** [comment]
I really like checkboxes in the task description as currently implemented. I think the subtasks may be useful, but they are different feature.

Could  you please provide few screenshots to show how other clients deal with the subtasks and how the subtasks are implemented (source code of the VTODO)?

**Comment by jakob42** [comment]
Seems to be just as the above linked rfc says. Created those with Nextcloud:

Main task:
```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//Nextcloud Tasks v0.9.5
BEGIN:VTODO
CREATED:20170630T095435
DTSTAMP:20170630T095507
LAST-MODIFIED:20170630T095507
UID:3o3al1t55r
SUMMARY:Subtask-Test
PRIORITY:0
PERCENT-COMPLETE:0
X-OC-HIDESUBTASKS:0
DESCRIPTION:Test task
END:VTODO
END:VCALENDAR
```

Sub task:
```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//Nextcloud Tasks v0.9.5
BEGIN:VTODO
CREATED:20170630T095507
DTSTAMP:20170630T095507
LAST-MODIFIED:20170630T095507
UID:dngfaro6wt8
SUMMARY:Subtask
PRIORITY:0
PERCENT-COMPLETE:0
X-OC-HIDESUBTASKS:0
RELATED-TO:3o3al1t55r
END:VTODO
END:VCALENDAR
```

**Comment by Matthias84** [comment]
Yep, this is how [nextcloud tasks app](https://github.com/nextcloud/tasks) looks like
![nextcloud tasks examples](https://user-images.githubusercontent.com/217511/27771796-c5466b54-5f55-11e7-961d-55b86fc6a0b3.jpg)

**Comment by dmfs** [comment]
FTR: opentask-provider already supports task relations, but there is no UI support yet.

**Comment by webermax** [comment]
hello @dmfs,

your work is awesome and the subtask feature as explained above would make open tasks feature complete far ahead of any other solution out there.

not only are you releasing FOSS but your clean and effective (gui) implementation is impressive.

i spent a lot of time evaluating existing solutions from evernote to todoist but the combination nextcloud/open tasks is the most flexible ON PREMISE solution existing (which even works in a cooperative manner).

please give me the possibility to motivate you continuing your great work, e.g. via a donation campaign.

**Comment by dmfs** [comment]
@webermax thanks! We're actually going to set up a small one-pager for OpenTasks and it will certainly have a donate button. At present we're focusing on getting recurrence support done. Although, we've also completed a few smaller tasks along the way (mostly minor UI/UX improvements).

Subtasks, custom alarms and categories are certainly the next big blocks once recurrence is out of the way.

**Comment by webermax** [comment]
@dmfs glad to hear that.

just checked http://dmfs.org/ and could not find any donate button yet.

looking forward seeing any news about your project anyway!

**Comment by unclev** [comment]
I wait for this feature too. Is there any work in progress on this or at least UI modifications ideas? From my point of view it looks as hard as replacing ListView with TreeView, isn't it?

**Comment by webermax** [comment]
@unclev I guess it is more then that. see https://github.com/dmfs/opentasks/pull/469
